clash-lib/src/Clash/Core/Evaluator/Models.hs

```diff
@@ -173,10 +159,12 @@ data GlobalEnv = GlobalEnv
   { genvGlobals :: VarEnv (Binding TermOrValue)
     -- ^ Global term environment. These are functions in global scope which
     -- are evaluated on lookup, and updated after evaluation.
-  , genvIgnore  :: Set Id
-    -- ^ Global ignore list. These are global identifiers which are not
-    -- inlined while evaluating. Items are added / removed during sections
-    -- where inlining would lead to non-termination.
+  , genvRecInfo :: RecInfo
+    -- ^ Information about which global bindings are recursive, used to decide
+    -- whether or not to inline a global binding.
```

Comment:
Should we store this on the binder itself? Does your discussion with @christiaanb change any of this?